### PR TITLE
Add CPU profiler test for Node.js

### PR DESCRIPTION
**Pull request by nsavoire**
* Allow to set per profile type filename regex.
 * Also check labels upon value check.
 * Interpret 0 percent/value as a check for non-existence
 * Allow regex in labels.
 * Rename node_wall to node_wall_cpu and add CPU test.
 * Make ddprof_allocations test less flaky by using a regex for thread/process id instead of hardcoded value

### CODE REVIEW

File: pprof_analysis.go [hunk 1]
```diff
@@ -46,6 +48,22 @@ type StackTestData struct {
 	Stacks   []TypedStacks `json:"stacks"`
 }
```
Comment (r1viollet) [inline pprof_analysis.go:50]: perhaps we can explain why this unmarshal is required (as default value is 0 and we use the 0 value to specify non-existance)

File: scenarios/node_wall_cpu/main.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+const {isMainThread, Worker, workerData} = require('worker_threads');
```
Comment (r1viollet) [inline scenarios/node_wall_cpu/main.js:1]: We should add a README to this test. it is complex

Comment (nsavoire) [reply scenarios/node_wall_cpu/main.js:1]: done

File: pprof_analysis.go [hunk 3]
```diff
@@ -405,10 +406,16 @@ func AnalyzeResults(t *testing.T, jsonFilePath string, pprof_folder string) {
...
 
-	// python files are in the form "profile.<pid>.number"
-	// Other profilers (using pprof) include pprof in the name
-	// Filter out files that ends with '.json' to avoid considering files dumped by captureProfData as profiles
-	default_pprof_regexp := regexp.MustCompile("(^profile.*|.*pprof.*)([^n]|[^o]n|[^s]on|[^j]son|[^.]json)$")
+	var default_pprof_regexp *regexp.Regexp
+	if stackTestData.PprofRegex != "" {
+		default_pprof_regexp = regexp.MustCompile(stackTestData.PprofRegex)
+	} else {
+		// python files are in the form "profile.<pid>.number"
```
Comment (r1viollet) [inline pprof_analysis.go:440]: good comment

**Review by r1viollet — APPROVED** [review]
LGTM

**Pull request merged**